src/portability/traits.md

```diff
@@ -0,0 +1,50 @@
+# The Trait System
+
+```rust
+/// Single digital push-pull output pin
+pub trait OutputPin {
+    /// Drives the pin low
+    fn set_low(&mut self);
+
+    /// Drives the pin high
+    fn set_high(&mut self);
+}
+```
+
+[rust-embedded/embedded-hal](https://github.com/rust-embedded/embedded-hal/blob/master/src/digital.rs)
+
+```rust
+impl<MODE> OutputPin for OutputGpio<MODE> {
+    fn set_low(&mut self) {
+        self.set_pin_low()
+    }
+
+    fn set_high(&mut self) {
+        self.set_pin_high()
+    }
+}
+```
+
+> this goes in your chip crate
```

Comment:
NOTE: Make the wording (and presentation) consistent - use "Device Crate", include links to all example crates, consistent structuring.